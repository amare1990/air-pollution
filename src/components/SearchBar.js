import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { fetchData } from '../redux/current/current';
import cityToGeoData from '../redux/geocodedata';
import '../styles/SearchBar.css';
import apiID from '../http-commons';
import { fetchGeoCode } from '../redux/city/city';

function SearchBar() {
  const [city, setCity] = useState('');
  const [result, setSearch] = useState('');
  const [lat, setLat] = useState('');
  const [lon, setLon] = useState('');
  const dispatch = useDispatch();

  const handleChange = (event) => {
    setCity(event.target.value);
  };

  let geoCodeURL = '';
  const handleSearch = async () => {
    setSearch(city);

    geoCodeURL = `http://api.openweathermap.org/geo/1.0/direct?q=${city}&limit=1&appid=${apiID}`;
    const geocode = await cityToGeoData(geoCodeURL);
    const { lat } = geocode[0];
    const { lon } = geocode[0];
    setLat(lat);
    setLon(lon);
  };
  useEffect(() => {
    if (lat && lon) {
      const urlCurrentPollutionData = 'http://api.openweathermap.org/data/2.5/air_pollution?'
        + `lat=${lat}&lon=${lon}&appid=${apiID}`;
      dispatch(fetchData(urlCurrentPollutionData));
      dispatch(fetchGeoCode(city));
    }
  }, [dispatch, geoCodeURL, lat, lon, city]);

  return (
    <>
      <div className="search-bar-wrapper">
        <input
          className="search-input"
          type="search"
          id={result}
          value={city}
          placeholder="search City"
          onChange={handleChange}
        />
        <button
          type="button"
          className="btn search-btn"
          onClick={handleSearch}
        >
          Search
        </button>
      </div>
    </>
  );
}

export default SearchBar;
