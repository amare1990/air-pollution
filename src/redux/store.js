import { configureStore } from '@reduxjs/toolkit';
import pollutionReducer from './current/current';
import cityReducer from './city/city';

export default configureStore({
  reducer: {
    pollution: pollutionReducer,
    city: cityReducer,
  },
});
