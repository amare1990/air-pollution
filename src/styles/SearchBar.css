.search-bar-wrapper {
  margin-top: 20px;
  background-color: #ddd;
  padding: 20px 20px 20px 10px;
}

.search-input,
.search-btn {
  height: 30px;
  border: 1px solid #0f0f0f;
  border-radius: 7px;
  padding: 2px 3px;
  margin-right: 3px;
}

.search-btn {
  width: 70px;
  background-color: aqua;
}

.search-btn:hover {
  background-color: steelblue;
}
