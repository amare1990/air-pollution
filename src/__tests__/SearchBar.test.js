import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SearchBar from '../components/SearchBar';
import store from '../redux/store';

test('Testing if SearchBar component renders correctly', () => {
  const searchBarSnapshot = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <SearchBar />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(searchBarSnapshot).toMatchSnapshot();
});
