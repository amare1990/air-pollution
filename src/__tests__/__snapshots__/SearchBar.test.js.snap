// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if SearchBar component renders correctly 1`] = `
<div
  className="search-bar-wrapper"
>
  <input
    className="search-input"
    id=""
    onChange={[Function]}
    placeholder="search City"
    type="search"
    value=""
  />
  <button
    className="btn search-btn"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
